### PR TITLE
Fix tests on Julia 0.7

### PR DESCRIPTION
**Pull request by nalimilan**
The remaining failure is fixed with DataStreams master.

### CODE REVIEW

File: src/abstractdataframe/abstractdataframe.jl [hunk 1]
```diff
@@ -606,7 +606,7 @@ julia> df
 ```
 """
 Base.filter!(f, df::AbstractDataFrame) =
-    deleterows!(df, findall(!f, eachrow(df)))
+    deleterows!(df, findall(collect(!f(r)::Bool for r in eachrow(df))))
```
Comment (ararslan) [inline src/abstractdataframe/abstractdataframe.jl]: Why is this change necessary? Because `eachrow` isn't indexable and `findall` requires an indexable collection?

File: src/other/utils.jl [hunk 2]
```diff
@@ -2,10 +2,13 @@ import Base: isidentifier, is_id_start_char, is_id_char
 
 const RESERVED_WORDS = Set(["begin", "while", "if", "for", "try",
     "return", "break", "continue", "function", "macro", "quote", "let",
-    "local", "global", "const", "abstract", "typealias", "type", "bitstype",
+    "local", "global", "const", "type",
     "immutable", "do", "module", "baremodule", "using", "import", "struct",
     "export", "importall", "end", "else", "elseif", "catch", "finally"])
 
+if VERSION < v"0.7.0-DEV.3220"
```
Comment (ararslan) [inline src/other/utils.jl]: Didn't this happen in 0.6-dev?

File: src/abstractdataframe/abstractdataframe.jl (hunk 1 above)
Comment (nalimilan) [reply src/abstractdataframe/abstractdataframe.jl]: Yes. That's kind of annoying, IIRC in the PR it was mentioned that a wrapper could be introduced in Base to use linear indices in such situations. But here it's probably faster to allocate a vector anyway (https://github.com/JuliaLang/julia/pull/25879), and we do that in `deleterows`.

File: src/other/utils.jl (hunk 2 above)
Comment (nalimilan) [reply src/other/utils.jl]: No, or the tests would fail on 0.6. This commit appears to be the one which removed these deprecated identifiers.

**Comment by bjarthur** [comment]
tests are failing again on 0.7-beta.  perhaps 0.7 should be added to travis.yml?

**Comment by nalimilan** [comment]
Sure.
